File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/mutate/RecordMutateDao.java
package ru.citeck.ecos.records3.record.op.mutate;

import org.jetbrains.annotations.NotNull;
import ru.citeck.ecos.records3.record.op.atts.RecordAtts;
import ru.citeck.ecos.records2.RecordRef;
import ru.citeck.ecos.records3.record.dao.RecordsDao;

public interface RecordMutateDao extends RecordsDao {

    @NotNull
    RecordRef mutate(@NotNull RecordAtts record);
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/dao/mutate/RecordMutateDao.kt
package ru.citeck.ecos.records3.record.dao.mutate

import ru.citeck.ecos.records3.record.atts.dto.LocalRecordAtts
import ru.citeck.ecos.records3.record.dao.RecordsDao
import kotlin.jvm.Throws

interface RecordMutateDao : RecordsDao {

    @Throws(Exception::class)
    fun mutate(record: LocalRecordAtts): String
}
